serialization-example/main.go

```diff
@@ -0,0 +1,172 @@
+}
+
+func basicBlockTaggedDataExample() {
+	fmt.Println(">>>>>>>> BasicBlock - TaggedData >>>>>>>>")
+
+	blockJSON := "{\"protocolVersion\":3,\"networkId\":\"10549460113735494767\",\"issuingTime\":\"1675563954966263210\",\"slotCommitmentId\":\"0x498bf08a5ed287bc87340341ffab28706768cd3a7035ae5e33932d9a12bb30940000000000000000\",\"latestFinalizedSlot\":21,\"issuerId\":\"0x3370746f30705b7d0b42597459714d45241e5a64761b09627c447b751c7e145c\",\"block\":{\"type\":0,\"strongParents\":[\"0x304442486c7a05361408585e4b5f7a67441c437528755a70041e0e557a6d4b2d7d4362083d492b57\",\"0x5f736978340a243d381b343b160b316a6b7d4b1e3c0355492e2e72113c2b126600157e69113c0b5c\"],\"weakParents\":[\"0x0b5a48384f382f4a49471c4860683c6f0a0d446f012e1b117c4e405f5e24497c72691f43535c0b42\"],\"shallowLikeParents\":[\"0x163007217803006078040b0f51507d3572355a457839095e572f125500401b7d220c772b56165a12\"],\"payload\":{\"type\":5,\"tag\":\"0x68656c6c6f20776f726c64\",\"data\":\"0x01020304\"},\"burnedMana\":\"180500\"},\"signature\":{\"type\":0,\"publicKey\":\"0x024b6f086177156350111d5e56227242034e596b7e3d0901180873740723193c\",\"signature\":\"0x7c274e5e771d5d60202d334f06773d3672484b1e4e6f03231b4e69305329267a4834374b0f2e0d5c6c2f7779620f4f534c773b1679400c52303d1f23121a4049\"}}"
+
+	basicBlock := &iotago.ProtocolBlock{}
+	api.JSONDecode([]byte(blockJSON), basicBlock)
+
+	basicBlock.IssuingTime = genesisTimestamp.Add(12 * time.Second)
+	b, _ := api.JSONEncode(basicBlock)
+	fmt.Println(prettier(b))
+
+	b, _ = api.Encode(basicBlock)
+	fmt.Println(b)
+
+	fmt.Printf("BlockID: %s %d\n\n", basicBlock.MustID().ToHex(), basicBlock.MustID().Index())
+}
+
+func validationBlockExample() {
+	fmt.Println(">>>>>>>> ValidationBlock >>>>>>>>")
+
+	blockJSON := "{\"protocolVersion\":3,\"networkId\":\"10549460113735494767\",\"issuingTime\":\"1675563954966263210\",\"slotCommitmentId\":\"0x498bf08a5ed287bc87340341ffab28706768cd3a7035ae5e33932d9a12bb30940000000000000000\",\"latestFinalizedSlot\":0,\"issuerId\":\"0x3370746f30705b7d0b42597459714d45241e5a64761b09627c447b751c7e145c\",\"block\":{\"type\":1,\"strongParents\":[\"0x304442486c7a05361408585e4b5f7a67441c437528755a70041e0e557a6d4b2d7d4362083d492b57\",\"0x5f736978340a243d381b343b160b316a6b7d4b1e3c0355492e2e72113c2b126600157e69113c0b5c\"],\"weakParents\":[\"0x0b5a48384f382f4a49471c4860683c6f0a0d446f012e1b117c4e405f5e24497c72691f43535c0b42\"],\"shallowLikeParents\":[\"0x163007217803006078040b0f51507d3572355a457839095e572f125500401b7d220c772b56165a12\"],\"highestSupportedVersion\":3,\"protocolParametersHash\":\"0x0000000000000000000000000000000000000000000000000000000000000000\"},\"signature\":{\"type\":0,\"publicKey\":\"0x024b6f086177156350111d5e56227242034e596b7e3d0901180873740723193c\",\"signature\":\"0x7c274e5e771d5d60202d334f06773d3672484b1e4e6f03231b4e69305329267a4834374b0f2e0d5c6c2f7779620f4f534c773b1679400c52303d1f23121a4049\"}}"
+
+	validationBlock := &iotago.ProtocolBlock{}
```

Comment:
Wouldn't it be better to build these structures with the builders rather than hardcoding them? If any field name changes or anything else, we'll have to update the JSON which is hard to do.
But we can also leave as-is.